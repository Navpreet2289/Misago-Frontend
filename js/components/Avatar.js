// @flow
import classNames from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'
import { url } from 'misago/constants'

const Avatar = ({ className, size, size2x, user }: AvatarProps) => {
  const _size = size || 100
  const _size2x = size2x || _size

  return (
    <img
      alt=""
      className={classNames('user-avatar', className)}
      height={_size}
      src={getSrcUrlForUserSize(user, _size)}
      srcSet={getSrcUrlForUserSize(user, _size2x)}
      width={_size}
    />
  )
}

Avatar.propTypes = {
  className: PropTypes.string,
  size: PropTypes.number,
  user: PropTypes.object
}

const getSrcUrlForUserSize = (user?: User, size: number): string => {
  if (!!user && user.id && user.avatars) {
    return findAvatarForSize(user.avatars, size)
  } else {
    return url.blank_avatar
  }
}

const findAvatarForSize = (avatars: Array<UserAvatar>, size: number): string  => {
  let avatar: UserAvatar = avatars[0]
  avatars.forEach((av: UserAvatar) => {
    if (av.size >= size) {
      avatar = av
    }
  })
  return avatar.url
}

export default Avatar