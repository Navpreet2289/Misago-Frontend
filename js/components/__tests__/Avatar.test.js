import React from 'react'
import renderer from 'react-test-renderer'
import Avatar from '../Avatar'

jest.mock('misago/constants', () => {
  return { url: { blank_avatar: 'blank_avatar.png' } }
})

const user = {
  id: 1,
  avatars: [
    { size: 200, url: '/avatars/200.png' },
    { size: 100, url: '/avatars/100.png' },
    { size: 10, url: '/avatars/10.png' }
  ]
}

test('<Avatar /> component renders', () => {
  const component = renderer.create(<Avatar />)
  expect(component).toMatchSnapshot()
})

test('<Avatar /> component renders with class name', () => {
  const component = renderer.create(
    <Avatar className="poster-avatar" />
  )

  expect(component).toMatchSnapshot()
})

test('<Avatar /> component renders for anonymous', () => {
  const anonymous = { id: 0 }
  const component = renderer.create(
    <Avatar user={anonymous} />
  )

  expect(component).toMatchSnapshot()
})

test('<Avatar /> component ignores size prop when no user is provided', () => {
  const component = renderer.create(
    <Avatar size={200} />
  )

  expect(component).toMatchSnapshot()
})

test('<Avatar /> component renders for user', () => {
  const component = renderer.create(
    <Avatar user={user} />
  )

  expect(component).toMatchSnapshot()
})

test('<Avatar /> component renders for user with size', () => {
  const component = renderer.create(
    <Avatar size={200} user={user} />
  )

  expect(component).toMatchSnapshot()
})

test('<Avatar /> component renders for user with size and size2x', () => {
  const component = renderer.create(
    <Avatar size={100} size2x={200} user={user} />
  )

  expect(component).toMatchSnapshot()
})