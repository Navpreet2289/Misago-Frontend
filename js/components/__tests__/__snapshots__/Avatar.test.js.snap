// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> component ignores size prop when no user is provided 1`] = `
<img
  alt=""
  className="user-avatar"
  height={200}
  src="blank_avatar.png"
  srcSet="blank_avatar.png"
  width={200}
/>
`;

exports[`<Avatar /> component renders 1`] = `
<img
  alt=""
  className="user-avatar"
  height={100}
  src="blank_avatar.png"
  srcSet="blank_avatar.png"
  width={100}
/>
`;

exports[`<Avatar /> component renders for anonymous 1`] = `
<img
  alt=""
  className="user-avatar"
  height={100}
  src="blank_avatar.png"
  srcSet="blank_avatar.png"
  width={100}
/>
`;

exports[`<Avatar /> component renders for user 1`] = `
<img
  alt=""
  className="user-avatar"
  height={100}
  src="/avatars/100.png"
  srcSet="/avatars/100.png"
  width={100}
/>
`;

exports[`<Avatar /> component renders for user with size 1`] = `
<img
  alt=""
  className="user-avatar"
  height={200}
  src="/avatars/200.png"
  srcSet="/avatars/200.png"
  width={200}
/>
`;

exports[`<Avatar /> component renders for user with size and size2x 1`] = `
<img
  alt=""
  className="user-avatar"
  height={100}
  src="/avatars/100.png"
  srcSet="/avatars/200.png"
  width={100}
/>
`;

exports[`<Avatar /> component renders with class name 1`] = `
<img
  alt=""
  className="user-avatar poster-avatar"
  height={100}
  src="blank_avatar.png"
  srcSet="blank_avatar.png"
  width={100}
/>
`;
